### DISCUSSION TITLE
Out of memory using 12gb 3060

### DISCUSSION THREAD

**Discussion by cosmicr** [d8ahazard/sd_dreambooth_extension#543] [Q&A]
I know this gets asked daily, but I'm at my witts end trying to get this working. So I apologise if this post isn't allowed.

I've looked at countless other forums and threads and can't for the life of me get it to work.

My settings:

```
Num examples = 151
  Num batches each epoch = 151
  Num Epochs = 125
  Instantaneous batch size per device = 1
  Total train batch size (w. parallel, distributed & accumulation) = 1
  Gradient Accumulation steps = 1
  Total optimization steps = 18875
  Actual steps: 18875
   Training settings: CPU: False Adam: True, Prec: bf16, Grad: True, TextTr: False EM: False, LR: 2e-06 LORA:False
 Allocated: 3.6GB
 Reserved: 3.7GB
```
Which always results in:
```
Steps:   0%|                                         | 0/18875 [00:00<?, ?it/s] Exception while training: CUDA out of memory. Tried to allocate 16.00 MiB (GPU 0; 12.00 GiB total capacity; 7.60 GiB already allocated; 2.48 GiB free; 7.72 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
```

Why does it say I have so little VRAM? I have a 12gb card. I have tried every different combination of settings, tried using Chrome, Edge, Firefox, closed every application on my computer from a fresh reboot. I can't even get it to run 1 epoch. I'm using windows 10, my startup batch file is:

```
@echo off

set PYTHON=
set GIT=
set VENV_DIR=
set ACCELERATE="False"
set COMMANDLINE_ARGS=--xformers
set "REQS_FILE=.\extensions\sd_dreambooth_extension\requirements.txt"

call webui.bat
```

It really depresses me to see people getting it working on 8gb cards!

**Comment by cerega66** [comment]
If you want to use a video card with 12GB or less in windows you need to set up wsl and run from it.

**Comment by CodeExplode** [comment]
I've been running it on a 3060 on windows using the same settings (didn't install wsl or anything). Yesterday I updated and had some OOM errors early on in the process a few times, then aren't sure why but it started working again (I think after restarting my PC).

Currently I think you need to train on 1.x model, and selecting half model might be important.

edit: I should also point out that I use xformers in the memory attention dropdown.

**Comment by mediocreatmybest** [comment]
I'm unsure if Flash_Attention works in Windows but have you tried that option? I for for the life of me haven't been able to get xformers working on my Linux box with a 3060 with training. It works no worries when I'm generating images etc, but as soon as I try xformers with the dreambooth training It crashes or throws CUDA errors.

**Comment by Thomas-MMJ** [comment]
Note you might want to uninstall and reinstall, I was having all sorts of weird issues and it seems to be working well once I uninstalled and reinstalled the sd_dreambooth extension

**Reply by jsbach-jung** [reply]
Yes, today I also had an unrelated weird issue after I updated directly from WebUI. When creating a new model, it would store string params as floats and so on; going back many commits still didn't help.

Finally I just did a clean install, and everything worked like nothing had happened.

**Comment by mediocreatmybest** [comment]
Sometimes a useful command that can reset the install to the latest head.

git reset --hard HEAD

It won't remove any additional files, but it will bring the install files back inline with the github commit.

**Reply by Thomas-MMJ** [reply]
Does that clear the pycache also?  I suspect pycache is the actual culprit.

**Reply by mediocreatmybest** [reply]
Nope, you would need to delete that manually. 
It is safe to delete as long as the program isn't running, it will re-cache itself.

Usually if I'm finding odd behavior, I'll delete the cache and reset the git with a --hard head to make sure its back to default.
